#IEEE-187: created new team detail page. populated page with table displaying te…

lukecheseldine opened the pull request:
## Overview

- Resolves #IEEE-187
- Created TeamDetail page, accessible at /team-detail
- Added table to display mock team data
- Styling in TeamDetail.module.scss


## Unit Tests Created

- renders user info table
- renders each user first and last name in the table
- renders each user email in the table
- renders each user phone number in the table
- users with id provided have a checkbox in the table


## Steps to QA

- Go to localhost:3000/team-detail

lukecheseldine commented:
one thing i know is wrong is that i'm using mui checkboxes, which are input elements. they have a hover and click effect, and also extra padding around them. do we have checkbox images that i can use instead of these checkbox input elements?

Leo6Leo commented on hackathon_site/dashboard/frontend/src/pages/TeamDetail/TeamDetail.module.scss:
```diff
@@ -0,0 +1,34 @@
+    }
+
+    .checkBox {
+        color: #2b7bbc;
```
.checkBox {
        margin-left: -15px;
        color: #2b7bbc;
    }

Add a margin so the title of the table will be on the same line as the checkbox.

Before:
<img width="618" alt="CleanShot 2022-05-29 at 16 00 07@2x" src="https://user-images.githubusercontent.com/36619969/170889253-c110c03e-aa32-4a04-863f-3333f79f2252.png">

After:
<img width="623" alt="CleanShot 2022-05-29 at 15 59 51@2x" src="https://user-images.githubusercontent.com/36619969/170889255-729afdbf-3302-400b-a32d-8b8315045fca.png">

Hope this will work on your divide. If not, we might need to think other approach.

YuYing-Liang replied:
Make sure you try this on different screen sizes, but if you want to remove the padding and margin, setting it to 0 in the css should work.

Leo6Leo reviewed (changes requested):
Overall Great job Luke!

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/App.tsx:
```diff
@@ -59,6 +60,11 @@ const UnconnectedApp = () => {
+                    <Route
+                        exact
+                        path="/team-detail"
+                        component={withUserCheck(TeamDetail)}
```
Sorry about this Luke, I told you the wrong setup. could you change this url to "/teams/:id" so that every team has their own page.

To get the /:id value you can use the useParams hook
e.g. `const { id } = useParams()`
in the team detail page file

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/pages/TeamDetail/TeamDetail.tsx:
```diff
@@ -0,0 +1,79 @@
+const TeamDetail = () => {
+    return (
+        <>
+            <Header />
```
You did a really good job with css! But usually, we try to stick with MUI components as much as we can and write custom css sparingly.
Here, I think it'd be a good place to use MUI Grid instead of the container and table container you have here (https://v4.mui.com/components/grid/)
and use the size breakpoints to test how it looks on different screens.
column sizes 6 for large screens and size 12 for mobile will work for this layout

YuYing-Liang reviewed (changes requested):
Nice job, great start!
Reminder to write unit tests to make sure everything is displaying properly.

lukecheseldine commented:
Refactored to use MUI Grid instead of custom css. 
Fixed minor css problems.
Created unit tests in TeamDetails.test.tsx to check:
- renders user info table
- renders each user first and last name in the table
- renders each user email in the table
- renders each user phone number in the table
- users with id provided have a checkbox in the table

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/pages/TeamDetail/TeamDetail.tsx:
```diff
@@ -0,0 +1,109 @@
+import React from "react";
+import styles from "./TeamDetail.module.scss";
```
Since you're not using the styling file anymore, you can remove the import. You can keep the file until the page is done in case we need it in the future.

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/pages/TeamDetail/TeamDetail.tsx:
```diff
@@ -0,0 +1,109 @@
+    Checkbox,
+    Paper,
+    Container,
+    Box,
```
you're not using Box so you can remove this too

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/pages/TeamDetail/TeamDetail.tsx:
```diff
@@ -0,0 +1,109 @@
+                                            {mockTeamMultiple.profiles.map((row) => (
+                                                <TableRow key={row.user.id}>
+                                                    <TableCell>
+                                                        {row.user.first_name}{" "}
```
```suggestion:
   {`${row.user.first_name} ${row.user.last_name}`}
```
Easier to handle whitespaces with ^^

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/pages/TeamDetail/TeamDetail.tsx:
```diff
@@ -0,0 +1,109 @@
+                                                        {row.user.phone}
+                                                    </TableCell>
+                                                    <TableCell>
+                                                        <Checkbox
```
You can use our predefined css variables instead of hardcoding the color (#2b7bbc is our primary colour, you can refer to it in this file: `src/assets/abstracts/_variables.scss` ), and you can use a self closing tag for the checkbox:
`
<Checkbox
         checked={row.id_provided}
         color="primary"
         style={{
              marginLeft: "-15px",
         }}
/>
`

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/pages/TeamDetail/TeamDetail.tsx:
```diff
@@ -0,0 +1,109 @@
+        <>
+            <Header />
+            <Container>
+                <Grid container direction="column" spacing={6}>
```
I'm confused why you need a Grid for this, <Typography variant="h1"> should be fine on its own for the title

YuYing-Liang replied:
Actually, I see what ur doing with the Grid and it's pretty nice 👍 , you can ignore this

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/pages/TeamDetail/TeamDetail.tsx:
```diff
@@ -0,0 +1,109 @@
+                            lg={6}
+                            xs={12}
+                        >
+                            <Grid item>
```
Same here, I don't think you need a nested grid for this, the only thing that needs a grid are the 2 tables.

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/pages/TeamDetail/TeamDetail.tsx:
```diff
@@ -0,0 +1,109 @@
+                            direction="column"
+                            spacing={1}
+                            item
+                            lg={6}
```
could you change this lg={6} to md={6}, that's our breakpoint for mobile view

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/pages/TeamDetail/TeamDetail.test.tsx:
```diff
@@ -0,0 +1,57 @@
+import { Provider } from "react-redux";
+
+describe("User info table", () => {
+    test("renders user info table", () => {
```
It's great you're breaking up the tests but you can combine all these unit tests into one because you're testing the same component. You can make all the assertions in a single forEach() loop instead of multiple (and it will save time when running tests :))

YuYing-Liang reviewed (commented):
Looks good Luke! Just some minor things

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/pages/TeamDetail/TeamDetail.test.tsx:
```diff
@@ -0,0 +1,43 @@
+
+        expect(screen.getByText("Team info")).toBeInTheDocument();
+
+        for (let i = 0; i < mockTeamMultiple.profiles.length; i++) {
```
nice job on all the tests! you don't need to change this but you can use the forEach() function here instead of a regular for loop
`mockTeamMultiple.profiles.forEach((profile, index) => {})`
javascript developers don't really use for loops anymore.

lukecheseldine replied:
ah i never knew i could easily access the index in forEach loops. ill do this in the future!

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/pages/TeamDetail/TeamDetail.tsx:
```diff
@@ -0,0 +1,106 @@
+                    </Grid>
+                    {/* Team info and actions */}
+                    <Grid container spacing={2}>
+                        <Grid
```
Could you create a new component for the team info table (it can include the grid elements too). The reason why is because this file is getting really big and we're not even done have the page yet 😅 to avoid bloated code, create a component in
`src/components/teamDetail/TeamInfoTable` with 2 files:
`TeamInfoTable.tsx` - you can name it whatever you want, just make sure it matches the folder name
`TeamInfoTable.test.tsx` - move your tests that only test what's displayed on the table to here
you don't have custom styling so you don't need to create a scss file.
Then, place the component here in TeamDetail.tsx.

YuYing-Liang reviewed (commented):
Just some small things but could you **only address them after this PR: https://github.com/ieeeuoft/hackathon-template/pull/397 is merged**

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/components/teamDetail/TeamInfoTable/TeamInfoTable.test.tsx:
```diff
@@ -0,0 +1,51 @@
+import TeamInfoTable from "components/teamDetail/TeamInfoTable/TeamInfoTable";
+import { RouteComponentProps } from "react-router-dom";
+
+const teamDetailProps = {
```
these props go towards the TeamDetail **page**, they're not needed for your table component so you can remove them

lukecheseldine replied:
@YuYing-Liang are you sure? i get the following error when i remove them in the test file. maybe im implementing the component incorrectly?
![Screen Shot 2022-06-12 at 6 42 44 PM](https://user-images.githubusercontent.com/36282235/173256511-9aaf4cb7-ee40-4b44-9783-2833ad84b0b4.png)
![Screen Shot 2022-06-12 at 6 43 07 PM](https://user-images.githubusercontent.com/36282235/173256509-0861684e-cfd3-4f1b-9ccc-bc8bb04f4df7.png)

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/components/teamDetail/TeamInfoTable/TeamInfoTable.tsx:
```diff
@@ -0,0 +1,107 @@
+import React from "react";
+import { PageParams } from "pages/TeamDetail/TeamDetail";
+
+export const TeamInfoTable = ({ match }: RouteComponentProps<PageParams>) => {
```
I think I may not have been clear, this component should only hold code for the **team info table**, not any other tables or headings. Some of this code (like the Team Overview title) is more associated with the page than the table.

Basically, everything from line 30 to 81 is the actual Team Info Table and the rest of the code should go back into TeamDetail.tsx

lukecheseldine replied:
fixed

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/pages/TeamDetail/TeamDetail.tsx:
```diff
@@ -0,0 +1,24 @@
+}
+
+const TeamDetail = ({ match }: RouteComponentProps<PageParams>) => {
+    const history = useHistory();
```
Based on the above comments, you shouldn't need to pass in any of the route page props since the TeamInfoTable component should only contain the table and not any other code.

lukecheseldine replied:
@YuYing-Liang see the above comment on the test file. basically whenever i write `<TeamInfoPage />` i get this error about missing props. this was my solution from a stackoverflow post

YuYing-Liang replied:
yeah thats because TeamInfoTable doesn't need these props, you can remove them from the component (see comments in TeamInfoTable.tsx)

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/components/teamDetail/TeamInfoTable/TeamInfoTable.tsx:
```diff
@@ -0,0 +1,65 @@
+
+export const TeamInfoTable = ({ match }: RouteComponentProps<PageParams>) => {
+    return (
+        <>
```
you do not need this tag (`<></>`), can be removed

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/components/teamDetail/TeamInfoTable/TeamInfoTable.tsx:
```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import { PageParams } from "pages/TeamDetail/TeamDetail";
+
+export const TeamInfoTable = ({ match }: RouteComponentProps<PageParams>) => {
```
remove these props and unused imports

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/pages/TeamDetail/TeamDetail.test.tsx:
```diff
@@ -0,0 +1,13 @@
+
+test("renders without crashing", () => {
+    render(<TeamInfoTable />);
+    expect(screen.getByText("Name")).toBeInTheDocument();
```
There was a test here that tested the team overview title was displaying properly, did you remove that?

YuYing-Liang replied:
Also ur rendering TeamInfoTable but this file should be for the TeamDetails page

YuYing-Liang reviewed (approved):
Looks great, good job on the team detail page test!

Leo6Leo reviewed (approved):
Yes Luke, it looks great to me! Great job!
